proto/provenance/expiration/v1/query.proto

```diff
@@ -0,0 +1,101 @@
+// Since: provenance 1.12
+syntax = "proto3";
+package provenance.expiration.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "provenance/expiration/v1/expiration.proto";
+// import "cosmos/base/v1beta1/coin.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+option go_package          = "github.com/provenance-io/provenance/x/expiration/types";
+option java_package        = "io.provenance.expiration.v1";
+option java_multiple_files = true;
+
+// Query defines the gRPC querier service for expiration module.
+service Query {
+  // Params queries the parameters for x/expiration
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/provenance/expiration/v1/params";
+  }
+
+  // Query Expiration
+  rpc Expiration(QueryExpirationRequest) returns (QueryExpirationResponse) {
+    option (google.api.http).get = "/provenance/expiration/v1/{module_asset_id}";
+  }
+
+  // Query all Expirations
+  rpc AllExpirations(QueryAllExpirationsRequest) returns (QueryAllExpirationsResponse) {
+    option (google.api.http).get = "/provenance/expiration/v1/all";
+  }
+
+  // Query Expirations by owner
+  rpc AllExpirationsByOwner(QueryAllExpirationsByOwnerRequest) returns (QueryAllExpirationsByOwnerResponse) {
+    option (google.api.http).get = "/provenance/expiration/v1/all/{owner}";
+  }
+
+  // Query all expired Expirations
+  rpc AllExpiredExpirations(QueryAllExpiredExpirationsRequest) returns (QueryAllExpiredExpirationsResponse) {
+    option (google.api.http).get = "/provenance/expiration/v1/all/expired";
```

Comment:
Have you checked that all of these REST endpoints work? I have a feeling it will be confused and think `params` and `all` are just `{module_asset_id}` values, and `expired` is an `{owner}` value. Sometimes things work better than I expect though.

Reply:
updated paths to no longer conflict.